pilot/pkg/networking/core/v1alpha3/name_table.go

```diff
@@ -98,7 +100,8 @@ func (configgen *ConfigGeneratorImpl) BuildNameTable(node *model.Proxy, push *mo
 						out.Table[host] = nameInfo
 					}
 
-					if instance.Endpoint.Locality.ClusterID != node.Metadata.ClusterID {
+					skipMulticluster := !features.MulticlusterHeadless && instance.Endpoint.Locality.ClusterID != node.Metadata.ClusterID
+					if skipMulticluster || !sameNetwork {
```

Comment:
The inclusion of `!sameNetwork` is new and is not flag protected. Is there any concern here?

Reply:
`skipMulticluster` comes from the original condition, so the old behavior should still _mostly_ work. 

Same network should be enforced regardless of the flag, since we'd never replace with gateway endpoints. .

Reply:
Ah ok. So you've added it to this condition just for clarity purposes?

Reply:
The original `clusterID` comparison was our only protection here against multi-network endpoints. It could be considered a bug since multi-network != multi-cluster